ea.go

```diff
@@ -28,6 +36,69 @@ type ExtendedAttribute struct {
 	Flags uint8
 }
 
+func GetFileEA(path string) ([]ExtendedAttribute, error) {
+	f, err := os.Open(path)
+	if err != nil {
+		return nil, fmt.Errorf("failed to open file: %s", err)
+	}
+	defer f.Close()
+
+	return GetFileEAHandle(windows.Handle(f.Fd()))
+}
+
+func GetFileEAHandle(handle windows.Handle) ([]ExtendedAttribute, error) {
```

Comment:
With the way the *Handle methods are named it seems like I'd expect to get back some handle to the EA's themselves which doesn't make sense (guess this is the way my brain works haha). Is exposing string and handle equivalents just to make this as flexible as possible?

Reply:
Do we really need the shortcut methods that take a file path? It's probably not too onerous to expect callers to pass a handle.

Reply:
Right, I think dropping the Handle off the names and having these be the only methods makes sense to me at least. Not the only method we have in this project that just takes in a handle and has no shortcut string equivalent https://github.com/microsoft/go-winio/blob/master/file.go#L113

Reply:
I added the method which takes the file path mainly for convenience and also so that we can encapsulate correct file opening permissions (for example, you need to open the file with specific access right when gettting/setting EA, or when gettting/setting reparse points). However, there are times when you already have a handle open for a file and you just want to pass in that (probably because you opened it in non sharing mode) that's why I added the other method. I agree that the handle methods can be named better but I think keeping both can be helpful. What do you think?

Reply:
@kevpar / @dcantah  any thoughts on this?

Reply:
What would you propose for the new names for the handle variants? I'm still leaning on just having the handle functions only. If the operations require specific permissions when opening the file, that can be called out in the comment for the function ("file handle must have X | Y | Z permissions for this function" or similar).

Reply:
I was thinking something like `GetEAforHandle` and `GetFileEA`. I am okay with having handle functions only but then everytime you call them you will have to get a handle to that file by doing `windows.CreateFile` calls with specific permissions. Anyway, I will remove the file functions for now and keep only the handle functions. We can add file functions later if required.

Reply:
I have updated the getter setter functions for EA to only take the handle now. For reparse points I think it makes more sense to directly take filepath as the input instead of taking the handle so I have kept those functions as it is. For the get security descriptor function there is not much value in adding that function if the caller also has to open the handle themselves, so I have kept that function too. Let me know what you guys think.

Reply:
If our use case for these is to open a handle, and then query SDs, EAs and reparse points it make sense to just have the NtAPI style that take handles. Otherwise there is a separate createfile on each function call.

Reply:
I think for reparse points its best to have the functions that take file paths but for EA we mostly deal with the handles so I have kept handles there. For SDs there is no particular preference but the wrapper from `golang.org/x/sys/windows` already  provides a function which takes in the handle so I kept the file path version here.

Reply:
Can you explain this comment a bit more? Why is it best to take paths for reparsepoint functions?

Reply:
I only said that because in order to get handle to a reparse point you need to open it with the `FLAG_OPEN_REPARSE_POINT` flag so its better to do that in the call that gets or sets the reparse points than expecting the caller to do it.

Reply:
@msscotb , @kevpar , @dcantah I have updated the PR, with the latest set of changes getters/setters for EAs take a file handle, getter for SDDL takes a path and getter for reparse takes a path but setter for reparse takes a file handle. The reason for this is that a reparse point could be created on a normal file or a directory so I have left the job of opening the handle to it on the caller. Let me know if this looks good.